lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -203,6 +203,30 @@ ResolveTypeAlias(SwiftASTContext *module_holder,
   return {n, {}};
 }
 
+std::string
+TypeSystemSwiftTypeRef::GetTupleElementName(lldb::opaque_compiler_type_t type,
+                                            size_t idx) {
+  using namespace swift::Demangle;
+  Demangler dem;
+  NodePointer node = TypeSystemSwiftTypeRef::DemangleCanonicalType(dem, type);
+  if (!node || node->getKind() != Node::Kind::Tuple)
+    return "";
+  if (node->getNumChildren() < idx)
+    return "";
+  NodePointer child = node->getChild(idx);
+  if (child->getNumChildren() != 1 &&
+      child->getKind() != Node::Kind::TupleElement)
+    return "";
+  for (NodePointer name : *child) {
+    if (name->getKind() != Node::Kind::TupleElementName)
+      continue;
+    return name->getText().str();
+  }
+  std::string name;
+  llvm::raw_string_ostream(name) << idx;
```

Comment:
What's the use case for returning the index if it has no name?

Reply:
This function is needed by TypeSystemSwiftTypeRef::GetChildCompilerTypeAtIndex().
The implicit name of an unnamed tuple element is 0:, 1:, ... . See https://github.com/apple/llvm-project/blob/46170cc2f5a9af3db7652ef119ecee0dbc78392d/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L7018